src/Acheve.TestHost/Routing/Tokenizers/PrimitiveParameterActionTokenizer.cs

```diff
@@ -15,27 +15,37 @@ public void AddTokens<TController>(TestServerAction action, TestServerTokenColle
 
             for (var i = 0; i < parameters.Length; i++)
             {
-                if ((parameters[i].ParameterType.IsPrimitive
-                    ||
-                    parameters[i].ParameterType == typeof(string)
-                    ||
-                    parameters[i].ParameterType == typeof(decimal)
-                    ||
-                    parameters[i].ParameterType == typeof(Guid))
-                    && !IgnoreHeader(parameters[i]))
-                {
-                    var tokenName = parameters[i].Name.ToLowerInvariant();
-                    var tokenValue = action.ArgumentValues[i].Instance;
-
-                    if (tokenValue != null)
+                if (!IgnoreHeader(parameters[i])) {
+
+                    if (IsPrimitiveType(parameters[i].ParameterType))
+                    {
+                        var tokenName = parameters[i].Name.ToLowerInvariant();
+                        var tokenValue = action.ArgumentValues[i].Instance;
+
+                        if (tokenValue != null)
+                        {
+                            tokens.AddToken(tokenName, tokenValue.ToString(), isConventional: false);
+                        }
+                    }
+                    else if (parameters[i].ParameterType.IsArray
+                       && IsPrimitiveType(parameters[i].ParameterType.GetElementType()))
                     {
-                        tokens.AddToken(tokenName, tokenValue.ToString(), isConventional: false);
+                        dynamic tokenValue = action.ArgumentValues[i].Instance;
```

Comment:
is "dynamic" mandatory? we try not to use it

Reply:
it was not mandatory to use it, but I have given it a workarround